### ISSUE TITLE
feat: Enhanced composite shader with Fresnel gloss and medium-specific rendering

### ISSUE THREAD

**Issue by abarth** [abarth/impression#122]
### Context
The current composite shader (`shaders/wet_media_composite.wgsl`) uses a simple `spec_strength = 0.3 * wetness` for specular highlights. This doesn't account for Fresnel effects (edges should be shinier), doesn't vary by medium type, and produces a uniform look regardless of whether the paint is oil or acrylic. Oil paint should remain somewhat glossy even when dry, while acrylic should dry to a matte finish.

### Goals
1. Replace simple specular with a physically-based Fresnel gloss model
2. Add medium-type-specific rendering behavior
3. Make the wet-to-dry gloss transition more realistic

---

### 4a. Enhanced Gloss Model

**File**: `shaders/wet_media_composite.wgsl`

Replace the current specular calculation (approximately lines 60-70):
```wgsl
// CURRENT:
let spec = pow(max(dot(normal, half_dir), 0.0), 32.0);
let spec_strength = 0.3 * wetness;

// NEW:
let roughness = mix(0.8, 0.1, wetness);  // dry=rough, wet=smooth
let view_dir = vec3f(0.0, 0.0, 1.0);     // camera looks down Z
let n_dot_v = max(dot(normal, view_dir), 0.0);
let fresnel = 0.04 + 0.96 * pow(1.0 - n_dot_v, 5.0);  // Schlick approximation
let spec_exponent = mix(8.0, 64.0, 1.0 - roughness);   // rough=broad, smooth=tight
let spec = pow(max(dot(normal, half_dir), 0.0), spec_exponent);
let spec_strength = fresnel * (1.0 - roughness * roughness);
```

### 4b. Medium-Type Branching

Add `medium_type: u32` to the composite uniform buffer (currently has `opacity` as bitcast u32).

Per-medium adjustments:
```wgsl
// Oil: stays somewhat glossy even when fully dry
if (medium_type == 0u) {  // Oil
    let min_gloss = 0.15;
    spec_strength = max(spec_strength, min_gloss * fresnel);
}
// Acrylic: dries to matte finish
else if (medium_type == 1u) {  // Acrylic
    let min_gloss = 0.02;
    spec_strength = max(spec_strength, min_gloss * fresnel);
}
// Watercolor: no specular (future)
else if (medium_type == 2u) {
    spec_strength = 0.0;
}
```

### 4c. Uniform Buffer Update

**File**: `src/gpu.ts`

The composite shader uniform currently passes only `opacity`. Expand to include `medium_type`:
- Update the uniform buffer size for wet media composite
- Write `medium_type` as a u32 alongside opacity
- Update the composite dispatch call in `compositor.ts` or wherever the bind group is created

The composite shader struct becomes:
```wgsl
struct CompositeParams {
    opacity_bits: u32,    // bitcast to f32
    medium_type: u32,     // 0=Oil, 1=Acrylic, 2=Watercolor
};
```

### 4d. Light Direction Refinement (Optional)

The current hardcoded light direction `normalize(vec3f(-0.4, -0.6, 0.8))` works but could be improved:
- Consider making height_scale (currently hardcoded at 4.0) configurable per-medium
- Oil typically has deeper impasto (height_scale = 5.0)
- Acrylic has moderate impasto (height_scale = 3.0)

This can be passed in the uniform buffer as well.

### Files to Modify

| File | Change |
|------|--------|
| `shaders/wet_media_composite.wgsl` | Fresnel specular, medium branching, roughness model |
| `src/gpu.ts` | Expand composite uniform buffer, pass medium_type |
| `src/compositor.ts` | Pass medium_type when creating composite bind group |

### Tests

This phase is primarily visual. Manual verification:
- **Oil wet stroke**: Bright specular highlights, glossy appearance
- **Oil dry stroke**: Reduced but still visible gloss (semi-glossy)
- **Acrylic wet stroke**: Similar bright specular to oil
- **Acrylic dry stroke**: Nearly matte, minimal specular
- **Fresnel at edges**: Impasto ridges should show brighter specular at glancing angles

### Verification
- Compare before/after screenshots of oil and acrylic strokes
- Verify impasto lighting still looks natural (not over-bright or flat)
- Confirm undo/redo works correctly (medium_type flows through brush settings)
- Performance: composite is a fragment shader, Fresnel adds ~2 instructions — negligible cost
